src/main/java/controllers/SwingBoard.java

```diff
@@ -0,0 +1,234 @@
+        this.state = state;
+
+        initNewGame();
+
+
+    }
+
+    public void initNewGame() {
+        log.info("New Game Generated");
+        initComponents();
+        while (state.getComputerShips().size() < state.getNoOfComputerShips()) {
+            initComputerShips(playerSelection.createComputerCoords());
+        }
+        printPanelCompPoints(mainPanel);
+
+    }
+
+
+    class Panel extends JPanel {
+        public void paintComponent(Graphics g) {
+            super.paintComponent(g);
+            g.drawRect(0, 0, 100, 100);
+        }
+    }
+
+
+    private void initComponents() {
+        int height = 800 * 2 / 3;
+        int width = 800 * 2 / 3;
+
+        frame = new JFrame("Battleship");
+        frame.setResizable(false);
+        frame.setPreferredSize(new Dimension(width, height));
+
+        frame.setDefaultCloseOperation(JFrame.EXIT_ON_CLOSE);
+
+
+        mainPanel = new JPanel();
+
+        mainPanel.setLayout(new GridLayout(10, 10));
+        for (int y = 0; y < LENGTH; y++) {
+            for (int x = 0; x < WIDTH; x++) {
+                panel[y][x] = new Panel();
+                panel[y][x].setPreferredSize(new Dimension(100, 100));
+                panel[y][x].setBorder(BorderFactory.createLineBorder(Color.black));
+                panel[y][x].addMouseListener(this);
+                panel[y][x].setBackground(new Color(0x0F95FF));
+                mainPanel.add(panel[y][x]);
+            }
+        }
+
+
+        footerPanel = new JPanel();
+        tfield = new JTextField(30);
+        tfield.setEditable(false);
+        footerPanel.add(tfield);
+
+
+        frame.setDefaultCloseOperation(JFrame.EXIT_ON_CLOSE);
+        frame.setLocationRelativeTo(null);
+
+        frame.add(mainPanel, BorderLayout.CENTER);
+        frame.add(footerPanel, BorderLayout.PAGE_END);
+
+
+        frame.pack();
+        frame.setVisible(true);
+    }
+
+
+    private void printPanelCompPoints(JPanel mainPanel) {
+
+        log.info("Panel Total : " + mainPanel.getComponentCount());
+    }
+
+
+    private void initComputerShips(int randomPanel) {
+        int randomPanelX = mainPanel.getComponent(randomPanel).getX();
+        int randomPanelY = mainPanel.getComponent(randomPanel).getY();
+        ComputerShip computerShip = (ComputerShip) playerSelection.createShip(randomPanelX, randomPanelY);
+        state.addComputerShips(computerShip);
+        System.out.println("X is: " + randomPanelX + "Y is: " + randomPanelY);
+
+
+    }
+
+
+    private void getWhichButtonGotPressed(int x, int y) {
+        tfield.setText("You Picked " + x + ", " + y);
+        //textbox displaying coordinates of the panel user clicked
+    }
+
+    private void displayEvent(int outcome) {
+        if (outcome == 0) {
+            JOptionPane.showMessageDialog(frame,
+                    "You hit the computer's ship!",
+                    "HIT!",
+                    JOptionPane.PLAIN_MESSAGE);
+            if (state.checkGameOver() == true) {
```

Comment:
Could this be simplified to `if (state.checkGameOver()) {`? There are a few similar cases in this class